### PR TITLE
added guidelines for logging

### PR DESCRIPTION
**Pull request by kaikreuzer**
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
I think this is very much in line with how we handle logging (levels) in the project, but I felt the necessity for formally write this down.

File: docs/documentation/development/guidelines.md [hunk 1]
```diff
@@ -51,3 +51,16 @@ Note that this list also serves as a checklist for code reviews on pull requests
...
 1. Bundles need to cleanly start and stop without throwing exceptions or malfunctioning. This can be tested by manually starting and stopping the bundle from the console (```stop <bundle-id>``` resp. ```start <bundle-id>```).
 1. Bundles must not require any substantial CPU time. Test this e.g. using "top" or VisualVM and compare CPU utilization with your bundle stopped vs. started.
+
+## E. Logging
+
+1. As we are in a dynamic OSGi environment, loggers should be [non-static](http://slf4j.org/faq.html#declared_static), when ever possible and have the name ```logger```.
+1. Parametrized logging must be used (instead of string concatenation).
+1. Where ever unchecked exceptions are caught and logged, the exception should be added as a last parameter to the logging. For checked exceptions, this is normally not recommended, unless it can be considered an error situation and the stacktrace would hold additional important information for the analysis.
```
Comment (maggu2810) [inline docs/documentation/development/guidelines.md]: as last parameter / argument without a `{}` placeholder in the string (don't know how to describe that correctly)

Comment (kaikreuzer) [inline docs/documentation/development/guidelines.md]: > don't know how to describe that correctly

Me neither, that's why I wrote it like that...

Comment (maggu2810) [inline docs/documentation/development/guidelines.md]: Hm, my comment was about adding that the for the exception / the last argument **no** placeholder (`{}`) should be added to the string...

Comment (kaikreuzer) [inline docs/documentation/development/guidelines.md]: I understand. As I didn't know how to formulate it, I didn't specifically mention it and hoped it is clear to anyone using slf4j anyhow. If you think something should be added, feel free to do so.

**Pull request merged**